### PR TITLE
Run update-ca-certificates before launching Rancher

### PR DESCRIPTION
**Pull request by thedadams**
The new SLE image requires update-ca-certificates be run before
additional trusted CAs are recognized. Therefore, the command is
run prior to launch Rancher.
    
In addition, the location of the mounted certificate is changed to the
location specified in the docs: /etc/pki/trust/anchors/

Finally, the jailer also copies the certificate information to /etc/ssl
instead of /etc/ssl/certs because the Go libraries expect the certs to
be there.

Issue:
https://github.com/rancher/rancher/issues/33398

### CODE REVIEW

File: package/entrypoint.sh
Comment (nickgerace) [inline package/entrypoint.sh:11]: I agree with the check here for local development, but shouldn't we fail if the executable is not found or in PATH?

Comment (thedadams) [reply package/entrypoint.sh:11]: If a user doesn't set `additionalTrustedCAs` to `true`, then we don't want to fail because `c_rehash` isn't on the `PATH`.

**Review by nickgerace — APPROVED** [review]

File: package/entrypoint.sh
Comment (jiaqiluo) [inline package/entrypoint.sh:9]: what if the command `c_rehash` exists but is not executable? Currently, we just ignore it.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: package/entrypoint.sh
Comment (thedadams) [reply package/entrypoint.sh:9]: The binary is included when we install `openssl` in the image, so I think it will be executable if it is there.

Comment (jiaqiluo) [reply package/entrypoint.sh:9]: Ok, so we do not need to concern about it.

**Review by jiaqiluo — DISMISSED** [review]
LGTM

**Comment by dirkmueller** [comment]
I suggest to read https://github.com/rancher/rancher/issues/33398#issuecomment-871163998 before merging this.

**Comment by thedadams** [comment]
The implementation has been updated according to @dirkmueller comment.

**Review by jiaqiluo — APPROVED** [review]

**Review by nickgerace — DISMISSED** [review]

File: chart/templates/deployment.yaml
Comment (cbron) [inline chart/templates/deployment.yaml:158]: Why are we removing the rancher one entirely ? Going back through old issues I see: 
> The difference in behaviour is the result of a change in the rancher deployment object in the helm chart in v2.2 vs 2.1, in which the custom CA file ca-additional.pem is now mounted at /etc/rancher/ssl/ca-additional.pem vs /etc/ssl/certs/ca-additional.pem (rancher/server-chart@7920ce4#diff-e3ed5b4174b5e628950137770a3e94d0). This change was made to rationalise certificate location into a single, non-OS dependent directory. However, pending updates to the git-based catalog design (15330) this breaks git-based custom catalogs from git servers presenting a cert signed with a custom CA.

So we were setting them in both places on purpose. 

If we do go with this change, it must be in the docs or guides. We'll have to open issue and find/replace.

Comment (superseb) [reply chart/templates/deployment.yaml:158]: The rancher one is there because we override the certificate location using ```ENV SSL_CERT_DIR /etc/rancher/ssl``` in the image. The other one is for the `git` binary.

Comment (thedadams) [reply chart/templates/deployment.yaml:158]: Now that we are using the SLE image, the [docs](https://documentation.suse.com/suse-caasp/4.5/html/caasp-admin/_troubleshooting_3.html#_error_x509_certificate_signed_by_unknown_authority) indicate that we should use the `/etc/pki/trust/anchors` directory. It seems that `update-ca-certificates` doesn't respect the `SSL_CERT_DIR` environment variable.

All that is to say that since we were setting the `SSL_CERT_DIR` to the custom directory and that is no longer being used, then we can remove it.

Comment (superseb) [reply chart/templates/deployment.yaml:158]: `SSL_CERT_DIR` is to override the system location for certificates when using Golang, so this can be removed if it is verified that Golang binaries (`rancher` it self but also `rancher-machine` still use the added certificates to validate connections)

Comment (thedadams) [reply chart/templates/deployment.yaml:158]: The `update-ca-certificates` command adds the certs in `/etc/pki/trust/anchors/` to `/etc/ssl/ca-bundle.pem` which is used by Go: https://golang.org/src/crypto/x509/root_linux.go

Comment (cbron) [reply chart/templates/deployment.yaml:158]: Did you test that docker machine still works with it ? I realize its also golang but we should do a smoke test.

I'm still not sold on this being all thats needed. How do the following still work: 
* Jailer: https://github.com/rancher/rancher/blob/master/package/jailer.sh#L46-L48
* RancherD (though this may be getting removed): https://github.com/rancher/rancher/blob/master/cmd/rancherd/chart/templates/daemonset.yaml#L98-L102
* RKE2: https://github.com/rancher/rancher/blob/master/pkg/controllers/provisioningv2/rke2/machineprovision/template.go#L193-L198

**Review by cbron — CHANGES_REQUESTED** [review]

File: package/jailer.sh
Comment (cbron) [inline package/jailer.sh:14]: > Finally, the jailer also copies the certificate information to /etc/ssl instead of /etc/ssl/certs because the Go libraries expect the certs to be there.

Confused why this change affects go libraries. This is all within the jail, the certs themselves are still being pulled out of the same place.

**Review by cbron — COMMENTED** [review]
One question, and also: 
* Still need to smoke test rancher-machine
* @dramich should approve jailer code

File: package/jailer.sh
Comment (thedadams) [reply package/jailer.sh:14]: The crypto package looks looks in different directories based on the OS: https://golang.org/src/crypto/x509/root_linux.go

Even though the run of `rancher-machine` is jailed, it will still look in the corresponding jailed directory for the certs.

**Comment by thedadams** [comment]
`rancher-machine` doesn't work: https://github.com/rancher/rancher/issues/33556

File: package/jailer.sh
Comment (cbron) [reply package/jailer.sh:14]: Ahh, so we are actually moving this because the crypto library looks in a different spot.

**Review by cbron — APPROVED** [review]
lgtm if my last assumption is correct

**Pull request merged**
